sparkplug-core/src/java/sparkplug/function/SerializableFn.java

```diff
@@ -21,6 +28,24 @@ private SerializableFn() {
     }
 
 
+    private void writeObject(ObjectOutputStream out) throws IOException {
+        // Attempt to derive the needed Clojure namespace from the function's class name.
+        String namespace = Compiler.demunge(this.f.getClass().getName()).split("/")[0];
+        out.writeObject(namespace);
+        out.writeObject(this.f);
+    }
+
+
+    private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
+        String namespace = (String)in.readObject();
+        Symbol namespaceSym = Symbol.intern(namespace);
+        synchronized (RT.REQUIRE_LOCK) {
```

Comment:
Worth adding a TODO comment above this to try an unsynchronized resolve first as a performance optimization.